lib/addons/ebs-csi-driver/index.ts

```diff
@@ -15,44 +17,121 @@ const versionMap: Map<KubernetesVersion, string> = new Map([
 /**
  * Interface for EBS CSI Driver EKS add-on options
  */
-export interface EbsCsiDriverAddOnProps {
-    /**
-     * Version of the driver to deploy
-     */
-    version?: string;
-    /**
-     * List of KMS keys to be used for encryption
-     */
-    kmsKeys?: kms.Key[];
+export interface EbsCsiDriverAddOnProps extends CoreAddOnProps {
+  /**
+   * List of KMS keys to be used for encryption
+   */
+  kmsKeys?: kms.Key[];
+  /**
+   * StorageClass to be used for the addon
+   */
+  storageClass?: string;
 }
 
 /**
  * Default values for the add-on
  */
-const defaultProps = {
-    addOnName: "aws-ebs-csi-driver",
-    version: "auto",
-    versionMap: versionMap,
-    saName: "ebs-csi-controller-sa",
+const defaultProps: CoreAddOnProps & EbsCsiDriverAddOnProps = {
+  addOnName: "aws-ebs-csi-driver",
+  version: "v1.23.0-eksbuild.1",
+  saName: "ebs-csi-controller-sa",
+  storageClass: "gp3", // Set the default StorageClass to gp3
 };
 
 /**
  * Implementation of EBS CSI Driver EKS add-on
  */
 @supportsALL
 export class EbsCsiDriverAddOn extends CoreAddOn {
+  readonly ebsProps: EbsCsiDriverAddOnProps;
 
-    constructor(readonly options?: EbsCsiDriverAddOnProps) {
-        super({
-            addOnName: defaultProps.addOnName,
-            version: options?.version ?? defaultProps.version,
-            saName: defaultProps.saName,
-            versionMap: defaultProps.versionMap,
-        });
-    }
+  constructor(readonly options?: EbsCsiDriverAddOnProps) {
+    super({
+      addOnName: defaultProps.addOnName,
+      version: options?.version ?? defaultProps.version,
+      saName: defaultProps.saName,

```

Comment:
same here: versionMap is removed. 
@elamaran11 FYI

Reply:
Corrected!